sub-packages/bionemo-llm/src/bionemo/llm/infer.py

```diff
@@ -0,0 +1,164 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+from pathlib import Path
+import pathlib
+from typing import Dict, Optional, Sequence, Type, cast, get_args
+
+import torch
+
+from bionemo.llm.run import config_models
+from bionemo.llm.model.biobert import model
+
+from bionemo.llm.run.config_models import DataConfig, ExperimentConfig, ExposedModelConfig, OptimizerSchedulerConfig, ParallelConfig, TrainingConfig, DataModuleT
+from bionemo.llm.utils.logger_utils import WandbConfig
+import nemo.lightning as nl
+from bionemo.llm.train import NsysConfig, setup_trainer 
+
+from bionemo.core.utils.dtypes import PrecisionTypes, get_autocast_dtype
+from bionemo.esm2.api import ESM2Config
+from bionemo.esm2.data.tokenizer import get_tokenizer
+from bionemo.esm2.model.finetune.datamodule import ESM2FineTuneDataModule, InMemoryCSVDataset
+from bionemo.llm.lightning import BionemoLightningModule, batch_collator
+from bionemo.llm.model.biobert.lightning import biobert_lightning_module
+from bionemo.llm.model.biobert.model import BioBertConfig
+from bionemo.llm.utils.datamodule_utils import infer_global_batch_size
+from bionemo.llm.run.config_models import ExModelConfigT, ModelConfigT
+
+__all__: Sequence[str] = ("infer",)
+
+def infer(
+    # core params
+    model_config_cls: type[ExModelConfigT | ModelConfigT],
+    checkpoint_path: str | pathlib.Path,
+    data_config: DataConfig[DataModuleT],
+    results_path: str | pathlib.Path,
+    # less interesting stuff
+    parallel_config: ParallelConfig,
+    precision: PrecisionTypes,
+    # Prediction parameters
+    include_hiddens: bool = False,
+    include_embeddings: bool = False,
+    include_logits: bool = False,
+    include_input_ids: bool = False,
+):
+
+    if isinstance(results_path, str):
+        results_path = pathlib.Path(results_path)
+
+    if os.path.isdir(results_path):
```

Comment:
This will change as we will require `results_path` to be a directory for storing results.